src/Example.SpecFlow/Hooks/Hooks1.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Windows.Forms;
+using ReportPortal.Client.Models;
+using ReportPortal.Client.Requests;
+using ReportPortal.SpecFlowPlugin;
+using ReportPortal.SpecFlowPlugin.EventArguments;
+using TechTalk.SpecFlow;
+
+namespace Example.SpecFlow.Hooks
+{
+    [Binding]
+    public sealed class Hooks1
+    {
+        [BeforeTestRun(Order = 0)]
+        public static void BeforeTestRunPart()
+        {
+            ReportPortalAddin.BeforeFeatureStarted += ReportPortalAddin_BeforeFeatureStarted;
+
+            ReportPortalAddin.BeforeScenarioStarted += ReportPortalAddin_BeforeScenarioStarted;
+            ReportPortalAddin.BeforeScenarioFinished += ReportPortalAddin_BeforeScenarioFinished;
+        }
+
+        private static void ReportPortalAddin_BeforeFeatureStarted(object sender, TestItemStartedEventArgs e)
+        {
+            // Adding feature tag on runtime
+            e.TestItem.Tags.Add("runtime_feature_tag");
+        }
+
+        private static void ReportPortalAddin_BeforeScenarioFinished(object sender, TestItemFinishedEventArgs e)
+        {
+            // Attaching screenshot at the end of the scenario
+            using (var bmpScreenCapture = new Bitmap(Screen.PrimaryScreen.Bounds.Width, Screen.PrimaryScreen.Bounds.Height))
+            {
+                using (var g = Graphics.FromImage(bmpScreenCapture))
+                {
+                    g.CopyFromScreen(Screen.PrimaryScreen.Bounds.X,
+                                     Screen.PrimaryScreen.Bounds.Y,
+                                     0, 0,
+                                     bmpScreenCapture.Size,
+                                     CopyPixelOperation.SourceCopy);
+                }
+
+                using (var stream = new MemoryStream())
+                {
+                    bmpScreenCapture.Save(stream, System.Drawing.Imaging.ImageFormat.Png);
+
+                    var attach = new Attach("attach", "image/png", stream.ToArray());
+
+                    e.Service.AddLogItem(new AddLogItemRequest
+                    {
+                        TestItemId = ReportPortalAddin.CurrentScenarioId,
+                        Level = LogLevel.None,
+                        Time = DateTime.UtcNow,
+                        Text = "Screenshot " + DateTime.Now,
+                        Attach = attach
+                    });
+                }
+            }
+        }
+
+        private static void ReportPortalAddin_BeforeScenarioStarted(object sender, TestItemStartedEventArgs e)
```

Comment:
I think we should not use nunit in customizations to show examples. Unit provider might be changes and this example will become incorrect.
Let's change this code to show:
* how to modify description
* how to add custom tags